### PR TITLE
Enforce LeftCurly rule

### PR DESCRIPTION
**Pull request by koppor**
This is a follow-up to https://github.com/JabRef/jabref/pull/6283.

I open this PR just for documentation. No need to review.

---

While reviewing https://github.com/JabRef/jabref/pull/6426, I found out, the contributors did not have setup the code style properly. Thus, I now put [LeftCurly](https://checkstyle.sourceforge.io/config_blocks.html#LeftCurly) in place, which enforces that there are no statements directly appended after an if.

```diff
-public DoubleProperty widthProperty() { return widthProperty; }
+public DoubleProperty widthProperty() {
+    return widthProperty;
+}
```

I am aware that for simple values, this causes more lines without improving things. Nevertheless, I vote for consistent code.

The **future** should be to rewrite minor code style issues (unused imports, line breaks, ...) automatically. The current tool of choise is [spotless](https://github.com/diffplug/spotless). Also [Google's refaster](https://errorprone.info/docs/refaster) should be put in place. It has *very* advanced rewriting templates. For instance:

```java
assertEquals(true, ...);
```

can be automatically rewritten to

```java
assertTrue(...);
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Argh, again some temporary issues:
         > Unable to load Maven meta-data from https://repository.apache.org/snapshots/org/apache/logging/log4j/log4j-jcl/3.0.0-SNAPSHOT/maven-metadata.xml.

File: src/main/java/org/jabref/gui/fieldeditors/FieldEditorFX.java [hunk 1]
```diff
@@ -25,11 +25,13 @@ default boolean childIsFocused() {
 
     /**
      * Returns relative size of the field editor in terms of display space.
-     *
-     * A value of 1 means that the editor gets exactly as much space as all other regular editors.
-     * A value of 2 means that the editor gets twice as much space as regular editors.
+     * <p>
+     * A value of 1 means that the editor gets exactly as much space as all other regular editors. A value of 2 means
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/fieldeditors/FieldEditorFX.java:29]: Minor point: Did you do this manually or via your code editor? The previous version was better in my opinion.

File: src/main/java/org/jabref/gui/maintable/MainTableColumnModel.java [hunk 2]
```diff
@@ -99,7 +99,8 @@ public MainTableColumnModel(Type type, String qualifier) {
     }
 
     /**
-     * This is used by the preferences dialog, to initialize available basic icon columns, the user can add to the table.
+     * This is used by the preferences dialog, to initialize available basic icon columns, the user can add to the
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/maintable/MainTableColumnModel.java:102]: same here, why the added line break?

File: src/main/java/org/jabref/gui/preferences/PreviewTabViewModel.java [hunk 3]
```diff
@@ -297,9 +299,8 @@ public void resetDefaultLayout() {
     }
 
     /**
-     * XML-Syntax-Highlighting for RichTextFX-Codearea created by (c) Carlos Martins (github: @cemartins)
-     * License: BSD-2-Clause
-     * see https://github.com/FXMisc/RichTextFX/blob/master/LICENSE and:
+     * XML-Syntax-Highlighting for RichTextFX-Codearea created by (c) Carlos Martins (github: @cemartins) License:
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/preferences/PreviewTabViewModel.java:302]: ...

**Comment by koppor** [comment]
I pressed automatic reformat in IntelliJ. No manual changes.

If we do not want to have these changes, we need to modify our **automatic** formatting rules.

File: src/main/java/org/jabref/gui/fieldeditors/FieldEditorFX.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/gui/fieldeditors/FieldEditorFX.java:29]: InteliJ autoformat. I added an additional <p>

It renders now as: (pressed <kbd>Ctrl</kbd>+<kbd>Q</kbd>)

grafik

Before, it rendered as 
grafik

**Comment by koppor** [comment]
This is a step forward that we are back in 2015, where we enforced that all contributors pressed <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>L</kbd> (in IntelliJ) to ensure proper formatting of the whole file.

File: config/IntelliJ Code Style.xml [hunk 4]
```diff
@@ -1,37 +1,35 @@
...
-    </value>
-  </option>
-  <option name="JD_P_AT_EMPTY_LINES" value="false" />
-  <option name="JD_KEEP_EMPTY_PARAMETER" value="false" />
-  <option name="JD_KEEP_EMPTY_EXCEPTION" value="false" />
-  <option name="JD_KEEP_EMPTY_RETURN" value="false" />
-  <option name="WRAP_COMMENTS" value="true" />
+<code_scheme name="JabRef" version="173">
+  <option name="RIGHT_MARGIN" value="140" />
+  <option name="WRAP_WHEN_TYPING_REACHES_RIGHT_MARGIN" value="true" />
```
Comment (tobiasdiez) [inline config/IntelliJ Code Style.xml:3]: I think we had a discussion about this some time ago and decided against having a hard line wrap (reasoning was "old school, everybody has wide monitors now anyway" if I remember correctly).

Comment (Siedlerchr) [reply config/IntelliJ Code Style.xml:3]: Yes, no hard-line wrap. I remember the discussion

**Comment by koppor** [comment]
I changed the maximum columsn from `120` to `140` at https://github.com/JabRef/jabref/pull/6452/commits/73f8ceb3dd45e5a6fea68554ba63b0741c5791e2.

- Google Java coding style: 100 characters. See https://qr.ae/pNynqB for a discussion. See also https://github.com/checkstyle/checkstyle/blob/4768470f54ec6f3031bd335157db9413c3906be9/src/main/resources/google_checks.xml#L45
- ChecksStyle code: 100 characters: See https://github.com/checkstyle/checkstyle/blob/4768470f54ec6f3031bd335157db9413c3906be9/config/checkstyle_checks.xml#L182.
- Sun: 80 characters. - This is the default of CheckStyle: https://checkstyle.sourceforge.io/config_sizes.html#LineLength

## Discussions in the net

- general discussion: https://dev.to/buinauskas/maximum-line-length-in-your-code-2ia4
- soft limit 120 chars: https://www.php-fig.org/psr/psr-12/
- more htan 100 chars: https://hackernoon.com/does-column-width-of-80-make-sense-in-2018-50c161fbdcf6

File: config/IntelliJ Code Style.xml (hunk 4 above)
Comment (koppor) [reply config/IntelliJ Code Style.xml:3]: I think, we did not adapt the config and we had no issues until I applied the automatic formatting today. Either none of use applied to automatic formatting before or was a wizzard in `git gui` only committing the reformat of code; not that of the comments.

I did the latter one often. However, currentyl, I believe, I cannot really explain to contributors that we have automatic tooling, but one cannot fully use it.

I changed the limit to 500 characters at https://github.com/JabRef/jabref/pull/6452/commits/59938a9606b3cce2013a799028564c2af3fdf75d. This should be high enough.

**Comment by koppor** [comment]
Filed an issue to IntelliJ, because it automatically reformats the JavaDoc even if not told so

https://youtrack.jetbrains.com/issue/IDEA-240517

File: config/IntelliJ Code Style.xml (hunk 4 above)
Comment (koppor) [reply config/IntelliJ Code Style.xml:3]: Removed the hard limit as we have files with 510 or 5000 characters line length.

**Comment by koppor** [comment]
Reformatted all files according to our IntelliJ formatting settings. I did not commit the change to one single line in the JavaDoc, but used the other changes.

Now all students can format the their contribution automatically.

For Eclipse, we need to check. Since these will be minor issues, I would propose to handle that as soon as an issue arises.

File: src/main/java/org/jabref/cli/JabRefCLI.java [hunk 5]
```diff
@@ -164,73 +164,73 @@ private static Options getOptions() {
 
         options.addOption(Option.builder("i").
                 longOpt("import").
-                desc(String.format("%s: '%s'", Localization.lang("Import file"), "-i library.bib")).
-                hasArg().
-                argName("FILE[,FORMAT]").
-                build());
+                                        desc(String.format("%s: '%s'", Localization.lang("Import file"), "-i library.bib")).
+                                        hasArg().
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/cli/JabRefCLI.java:168]: Here

**Pull request merged**